### PR TITLE
Uhf gnuradio interface + build fix

### PR DESCRIPTION
**Pull request by jlazzz**
Changes identical to PR #78 Uhf gnuradio interface, with minor additional changes made to ensure working build.

Note that working build depends on PR #25 on ex2_uhf_software to be approved and merged with master.

### CODE REVIEW

**Comment by joshdellaz** [comment]
@Taylorr82 fixed and rebased

File: src/groundStation/uTransceiver.py [hunk 1]
```diff
@@ -0,0 +1,125 @@
...
+class uTransceiver(object):
+
+    def __init__(self, opt):
+        self.listentimeout_s = 2.0
+        self.pipetimeout_s = 30.0
+        self.last_tx_time = 0
+        self.listen_en = False
+        self.pipe_en = False
+        self.u = opt
+        self.uhf = cdll.LoadLibrary("./ex2_uhf_software/uTransceiver.so")#consider making an env var for this?
```
Comment (Taylorr82) [inline src/groundStation/uTransceiver.py:38]: yes, do the comment

File: src/groundStation/uTransceiver.py [hunk 2]
```diff
@@ -0,0 +1,125 @@
...
+        #for testing purposes
+        # s = socket.socket(socket.AF_INET, socket.SOCK_STREAM)
+        # s.connect(("127.0.0.1", port))
+
+        print('Received from UHF:')
+        self.listen_en = True
+        start = time.time()
+        while (time.time() - start) < self.listentimeout_s:
+
+            #for testing purposes
```
Comment (Taylorr82) [inline src/groundStation/uTransceiver.py:64]: Debug code shouldn't be left commented out

**Review by Taylorr82 — CHANGES_REQUESTED** [review]

File: package.json [hunk 3]
```diff
@@ -18,6 +18,7 @@
     "csp:install": "cd libcsp &&  python3 waf install",
     "full_build": "yarn install_dependencies && yarn csp:clone && yarn build",
     "cli": "LD_LIBRARY_PATH=./libcsp/build PYTHONPATH=./libcsp/build python3 src/cli.py",
+    "cli_gdb": "LD_LIBRARY_PATH=./libcsp/build PYTHONPATH=./libcsp/build gdb python3",
```
Comment (Taylorr82) [inline package.json:21]: are you missing a python script?

File: src/groundStation/groundStation.py [hunk 4]
```diff
@@ -304,7 +307,7 @@ def handlePipeMode(self):
             if (time.time() - self.uTrns.last_tx_time) > self.uTrns.pipetimeout_s:
                 self.uTrns.enterPipeMode()
                 #may need to add delay here?
-                command, port, toSend = self.getInput(inVal='obc.general.UHF_IS_IN_PIPE_NOTIFICATION(1)')
+                command, port, toSend = self.getInput(inVal='ex2.general.UHF_IS_IN_PIPE_NOTIFICATION(1)')
```
Comment (Taylorr82) [inline src/groundStation/groundStation.py:310]: This should use self.satellite

File: src/groundStation/uTransceiver.py [hunk 5]
```diff
@@ -116,7 +117,7 @@ def UHFDIRCommand(self, string):
                 #TODO: make more robust handler for incorrect inputs to prevent erronious commands being sent?
                 if retval != 0:
                     print('UHF Equipment Handler error ' + UHF_return(retval).name)
-
+                time.sleep(0.5)
```
Comment (Taylorr82) [inline src/groundStation/uTransceiver.py:120]: Why are we sleeping here?

**Review by Taylorr82 — CHANGES_REQUESTED** [review]

File: package.json (hunk 3 above)
Comment (joshdellaz) [reply package.json:21]: it appears that way. I actually am having trouble getting gdb to work again, and I don't see it as necessary here, so I'm removing this line from the package.

**Comment by joshdellaz** [comment]
Rebased with master. All comments should be addressed.

**Review by jmmabanta — APPROVED** [review]
Looks good to me from a website integration point of view!

File: src/groundStation/uTransceiver.py [hunk 6]
```diff
@@ -0,0 +1,130 @@
...
+import time
+from enum import Enum
+import os
+
+# command format follows https://docs.google.com/spreadsheets/d/1zNhxhs0KJCp1187Vm3-zAzQHCY31f77l-0nlQmfXu1w/edit#gid=565953736
+# but with UHFDIR_ instead of UHF_ prefix. See document for examples
+
+#After sending a uhf-direct command, terminal displays all rx'd raw ASCII data for listentimeout_s
+
+class UHF_return(Enum):
```
Comment (Taylorr82) [inline src/groundStation/uTransceiver.py:17]: Add this to the python3 bindings

Comment (Taylorr82) [inline src/groundStation/uTransceiver.py:17]: See if you can access this enum from the library or something. Or we can create python bindings

**Pull request merged**